core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncResponseTransformer.java

```diff
@@ -118,6 +118,26 @@
         return new FileAsyncResponseTransformer<>(path);
     }
 
+    /**
+     * Creates an {@link AsyncResponseTransformer} that writes all the content to the given file.
+     *
+     * @param path            Path to file to write to.
+     * @param position        The value for the data between the current end of the file and the starting position is
+     *                        undefined.
+     * @param isNewFile       Whether this is a new file. If this is {@code true} and the file already exists, the
+     *                        transformer will complete with an exception.
+     * @param deleteOnFailure Whether the file on disk should be deleted in the event of a failure when writing the
+     *                        stream.
+     * @param <ResponseT> Pojo Response type.
+     * @return AsyncResponseTransformer instance.
+     */
+    static <ResponseT> AsyncResponseTransformer<ResponseT, ResponseT> toFile(Path path,
```

Comment:
note: I have another PR for this in https://github.com/aws/aws-sdk-java-v2/pull/1222 that will be merged to `master`. I will update the `s3-transfermanager` branch once it goes through. Just didn't want it to hold this PR up.